src/python/pants/backend/python/subsystems/python_setup.py

```diff
@@ -30,9 +30,10 @@ def register_options(cls, register):
     register('--interpreter-constraints', advanced=True, default=['CPython>=2.7,<3'], type=list,
              metavar='<requirement>',
              help="Constrain the selected Python interpreter.  Specify with requirement syntax, "
-                  "e.g. 'CPython>=2.7,<3' or 'PyPy'. Multiple constraints will be ORed together. "
-                  "These constraints are applied in addition to any compatibilities required by "
-                  "the relevant targets.")
+                  "e.g. 'CPython>=2.7,<3' (A CPython interpreter with version >=2.7 AND version <3)"
+                  "or 'PyPy' (A pypy interpreter of any version). Multiple constraints will be "
```

Comment:
Suggestion: may be worth adding `multiple constraint strings` instead of `multiple constraints` to remove any ambiguity around whether an entire string or just a simple inequality constitutes a "constraint".